프로젝트 기본 설정

ramram1048 opened the pull request:
### 참고
- 백엔드 express app webpack/babel 설정: 9f2f8c7b6b17240aeef7a1caaac2e9e924c3d5e7
- 프론트엔드 react app webpack/babel 설정: a90ee3bf5a1ebe0bdae3c9e7f67517f67b24164e

### 프로젝트 실행 방법
초기 fetch & pull
```bash
git fetch
git checkout feat-initial
git pull
npm i
```

백엔드 express app을 development environment로 build
```bash
npm run watch
```

백엔드 express app 실행
```bash
node dist/app.js
```

이후 `localhost:3000` ㄱㄱ

### 폴더 구조
```
.
├── /src
|   ├── /backend
|   |   ├── .eslintrc.json
|   |   ├── babel.config.js
|   |   ├── webpack.config.js
|   |   ├── app.js
|   |   └── devserver.js
|   └── /frontend
|       ├── components
|       |   ├── /Login
|       |   |   ├── Login.jsx
|       |   |   └── Login.scss
|       |   └── App.jsx
|       ├── resources
|       |   └── index.html
|       ├── .eslintrc.json
|       ├── babel.config.js
|       ├── webpack.config.js
|       └── index.js
├── package-lock.json
├── package.json
└── .gitignore
```

ramram1048 committed:
feat: express app이 /public 폴더를 static serve

ramram1048 committed:
chore: 개발용 react-hot-reload 적용

- NODE_ENV가 development일 때 express app에서
  webpack middleware로 실시간 build & serve
- frontend webpack 설정 세분화

ramram1048 committed:
chore: watch script 삽입

- 개발 시 backend webpack watch용 script

ramram1048 committed:
chore: scss rule 추가

ramram1048 committed:
feat: 테스트용 component 추가

- style 적용 및 컴포넌트 import 테스트용 Login 컴포넌트 추가
- App.js -> App.jsx로 확장자 변경

ramram1048 commented on src/backend/app.js:
```diff
@@ -1,11 +1,12 @@
-app.get('/', (req, res) => {
-  res.send('Hello World!');
-});
+if (process.env.NODE_ENV === 'development') useDevServer(app);
```
하하하

ramram1048 commented on src/frontend/components/Login/Login.jsx:
```diff
@@ -0,0 +1,10 @@
+import './Login.scss';
+import React from 'react';
+
+const Login = () => (
+  <div className='login'>
+    Login~~
+  </div>
+);
```
여러개

leejiwoo2002 reviewed (commented):
watch 옵션을 사용해서 개발시에 변경 사항을 라이브로 적용하도록 하신 점 확인했습니다

ramram1048 committed:
fix: `jsx` 확장자 import 관련 에러 수정

- jsx를 확장자 없이 import하는 기능은 철회
- component 폴더에 index.js를 만들어서 scss import와 jsx export를 담당하도록 구성

ramram1048 commented:
![image](https://user-images.githubusercontent.com/48747221/97262470-3cc0da00-1864-11eb-90bc-a56e0d87475a.png)
